addons/website_sale/models/mail_compose_message.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import datetime
+from odoo import api, fields, models
+
+class MailComposeMessage(models.TransientModel):
+    _inherit = 'mail.compose.message'
+
+    @api.multi
+    def send_mail(self, auto_commit=False):
+        context = self._context
+        if self.model == 'sale.order':
+            one_hour_before = datetime.datetime.utcnow() - datetime.timedelta(hours=1)
+            abandoned_cart_domain = [
+                ('id', 'in', context.get('active_ids')),
+                ('cart_recovery_email_sent', '!=', True),
+                ('state', '=', 'draft'),
+                ('partner_id.id', '!=', self.env.ref('base.public_partner').id),
+                ('order_line', '!=', False),
+                ('team_id.team_type', '=', 'website'),
+                ('date_order', '<', fields.Datetime.to_string(one_hour_before))]
+            abandonned_orders = self.env['sale.order'].search(abandoned_cart_domain)
+            abandonned_orders.write({'cart_recovery_email_sent': True})
+            self = self.with_context(mail_post_autofollow=True)
```

Comment:
Is it necessary to add the autofollow mechanism ? If you do it in batch it means you will be follower of all sale orders.